.gitignore

```diff
@@ -14,3 +14,6 @@ packages
 
 # Or local files with secrets
 key.json
+
+# Just for pkg/popularity
+popularity/pubspec.lock
```

Comment:
Why not `pkg/popularity/pubspec.lock` then?

Reply:
If a package is meant to be consumed – not an app – we generally don't include the lock file.

...the pubspec.lock file we really care about is in the pub site.